src/domain/relayer/evm/evm_relayer.rs

```diff
@@ -236,9 +236,51 @@ where
     /// # Returns
     ///
     /// A `Result` containing a boolean indicating the status or a `RelayerError`.
-    async fn get_status(&self) -> Result<bool, RelayerError> {
-        println!("EVM get_status...");
-        Ok(true)
+    async fn get_status(&self) -> Result<RelayerStatus, RelayerError> {
+        let relayer_model = &self.relayer;
+
+        let nonce_u256 = self
+            .provider
+            .get_transaction_count(&relayer_model.address)
+            .await
+            .map_err(|e| RelayerError::ProviderError(format!("Failed to get nonce: {}", e)))?;
+        let nonce_str = nonce_u256.to_string();
+
+        let balance_response = self.get_balance().await?;
+
+        let pending_statuses = [
+            TransactionStatus::Pending,
+            TransactionStatus::Submitted,
+            TransactionStatus::Mined,
+        ];
+        let pending_transactions = self
+            .transaction_repository
+            .find_by_status(&relayer_model.id, &pending_statuses[..])
+            .await
+            .map_err(RelayerError::from)?;
+        let pending_transactions_count = pending_transactions.len() as u64;
+
+        let confirmed_statuses = [TransactionStatus::Confirmed];
+        let confirmed_transactions = self
+            .transaction_repository
+            .find_by_status(&relayer_model.id, &confirmed_statuses[..])
```

Comment:
In one of PRs I have mentioned we could modify find_by_status to work on relayer level by default because we would rarely need to fetch all existing txs in system for all relayers.

Also we can include default sort.

If we do that this logic would be simpler for this PR as well.

Reply:
looks good as. I was thinking for some reason we are filtering here by relayer id.